### ISSUE TITLE
Precompiling LightGraph fails on Julia 0.7

### ISSUE THREAD

**Issue by ufechner7** [JuliaGraphs/LightGraphs.jl#942]
```
(v0.7) pkg> precompile
Precompiling project...
Precompiling ForwardDiff [1 of 12]
Precompiling LightGraphs [2 of 12]
ERROR: LoadError: ArgumentError: Package LightGraphs does not have IterativeEigensolvers in its dependencies:
- If you have LightGraphs checked out for development and have
  added IterativeEigensolvers as a dependency but haven't updated your primary
  environment's manifest file, try `Pkg.resolve()`.
- Otherwise you may need to report an issue with LightGraphs
Stacktrace:
 [1] require(::Module, ::Symbol) at ./loading.jl:832
 [2] include at ./boot.jl:317 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1050
 [4] include(::Module, ::String) at ./sysimg.jl:29
 [5] top-level scope at none:0
 [6] eval at ./boot.jl:319 [inlined]
 [7] eval(::Expr) at ./client.jl:394
 [8] top-level scope at ./none:3 [inlined]
 [9] top-level scope at ./<missing>:0
in expression starting at /home/ufechner/.julia/packages/LightGraphs/5nPh/src/LightGraphs.jl:9
ERROR: Failed to precompile LightGraphs to /home/ufechner/.julia/compiled/v0.7/LightGraphs/Xm08.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at ./logging.jl:298 [inlined]
 [3] compilecache(::Base.PkgId) at ./loading.jl:1189
 [4] _require(::Base.PkgId) at ./loading.jl:958
 [5] require(::Base.PkgId) at ./loading.jl:854
 [6] macro expansion at ./logging.jl:299 [inlined]
 [7] require(::Module, ::Symbol) at ./loading.jl:836
```

**Comment by sbromberger** [comment]
This is the same problem as #941. The only non-trivial workaround is to use a nightly build that has been compiled with the same version of gcc/clang as the Arpack binary.

**Comment by ufechner7** [comment]
This issue is not fixed, so why do you close it?

**Comment by juliohm** [comment]
@ufechner7 the issue is not in LG itself, but the interplay between Julia v0.7 and ARPACK.

It has been problematic to everyone. The easiest solution is to download the latest Julia binary (the nightly build). If you need a build of Julia from source, then there is no solution available that is easy. Just the one that @sbromberger mentioned.

**Comment by sbromberger** [comment]
As @juliohm said, this is not something we can fix in LightGraphs without removing spectral graph functions that depend on `eigs`. The problem is with Julia moving `eigs` out to a third-party repository where the version of compiler used for `Arpack` is different from the version used to (locally) compile the Julia binary. See https://github.com/JuliaLinearAlgebra/Arpack.jl/issues/5 for more details.

**Comment by jpfairbanks** [comment]
@ufechner7 The background for this is https://github.com/JuliaLinearAlgebra/Arpack.jl/issues/5
If you follow the instructions at https://github.com/JuliaLinearAlgebra/Arpack.jl/issues/5#issuecomment-404934563 you should be able to build ARPACK from source which would enable using LG with julia-master.

It would be very beneficial to provide a script for building ARPACK with LG.

**Comment by sbromberger** [comment]
Because before `eigs` was moved out to Arpack, it was in `IterativeEigensolvers` and you're probably running a version of LightGraphs that referred to that. If you're not on current master with Julia 0.7, you should be; if you are and you're still getting that error then it's a problem (but not a big one because it's broken in any case and fixing that issue won't resolve the larger issue.)

**Comment by ufechner7** [comment]
Using LightGraphs.jl#master on Ubuntu 16.04 with gcc 7.2 works fine. Remaining issues: 
1. create a released version that works on 0.7
2. add the info to Readme.md that gcc 7.2 is needed for LightGraphs 0n 0.7, compiled from source.

**Comment by sbromberger** [comment]
> create a released version that works on 0.7

This will happen once 0.7 is released. There have been too many changes in the last month to warrant making a new release now.

> add the info to Readme.md that gcc 7.2 is needed for LightGraphs on 0.7, compiled from source.

This might be a good notification.

**Comment by mlubin** [comment]
I'm running into this issue on travis (https://travis-ci.org/mlubin/ReverseDiffSparse.jl/jobs/409174252#L783). Is that expected with the latest release of LightGraphs?

**Comment by simonschoelly** [comment]
I think I had the same problem before and I think the reason is that travis uses the latest tagged version of LightGraphs but the newest changes in LightGraphs are not yet in this tagged version and therefore it is not compatible with the nightly version of Julia.

I 'fixed' it by explicitly downloading the latest version of LightGraphs in my .travis.yml file: 
```
language: julia
julia:
    - nightly

script:
    - julia -e 'Pkg.clone("https://github.com/JuliaGraphs/LightGraphs.jl.git")'
    - if [ -f test/runtests.jl ]; then julia --check-bounds=yes -e 'Pkg.test("ImplicitGraphs", coverage=true)'; fi

after_success:
    - julia -e 'cd(Pkg.dir("ImplicitGraphs")); Pkg.add("Coverage"); using Coverage; Codecov.submit(Codecov.process_folder())'

```

**Comment by sbromberger** [comment]
> Is that expected with the latest release of LightGraphs?

@mlubin Unfortunately, yes. At this time the latest tagged version of LG (0.13.1) doesn't work with -nightly, since it still references `IterativeEigensolvers`. I cannot test the new version that uses `Arpack` because I cannot get `Arpack.jl` to build in any of my environments, so if we were to tag 0.13.2, it'd be without my verification, which is a deviation from all other releases (I manually verify each release before tagging.)

We're in the middle of trying to figure out the best way to proceed here, but the fact that `eigs` was moved 1) out of base, 2) out of stdlib, and 3) into a package that *DOESN'T BUILD ON MACOS WHEN YOU BUILD JULIA FROM SOURCE* is a major source of frustration for me and is really bringing me closer to abandoning Julia entirely for a language that doesn't play so fast and loose with dependencies.

**Comment by sbromberger** [comment]
@ufechner7 we've just released LG v0.14.0, which explicitly requires Arpack. This won't help you if you're building Julia from source with a compiler that's not GCC7, but if you've gotten things to work with LG master and your Julia installation you now have a tagged version that should continue to work.
